src/database/PostgreSQL.ts

```
@@ -42,6 +42,10 @@ export class PostgreSQL implements IDatabase {
     });
   }
 
+  async initialize(): Promise<void> {
+
+  }
```

Are we sure that the empty body is sufficient here?

---

> > Is this getting complicated for the wrong reasons? I don't know enough but my code smell sense is going off.
> 
> Getting complicated in which way? Async is traditionally not very straightforward, and this is a nodejs web app. Having said that, is there anything in particular that looks especially complicated?

Good question. Well, what problem is the app currently exhibiting? I reported a startup issue (#2802) which we then only saw appeared with the filesystem database (if my recollection is correct), and so closed the bug. Are you saying that the issue also appears in other databases?

---

> Good question. Well, what problem is the app currently exhibiting? I reported a startup issue (#2802) which we then only saw appeared with the filesystem database (if my recollection is correct), and so closed the bug. Are you saying that the issue also appears in other databases?

The issues it aims to resolve are on the sidebar of this page those are #3500 and #3507.

---

This change is ensuring that database tables are created before we query them. The bug that appeared in #3500 was a symptom of a larger problem. If you delete the SQLite database file and run `npm run start` we have an error that goes unreported. 

https://github.com/bafolts/terraforming-mars/blob/main/src/database/SQLite.ts#L49

We ignore the error there and only call the callback if there are rows. If you modify this to not ignore the error you will see the error logged. For now I only made the change for SQLite hoping to minimize how much was changing with one PR.

I don't have to use async. I originally was going a different way with this PR and figured using promises would make this simpler. I can make `initialize` utilize a callback like we do elsewhere.

---

> I originally was going a different way with this PR and figured using promises would make this simpler. 

I agree with this.